android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -199,6 +202,12 @@ public void onStart() {
         }
     }
 
+    @Override
+    public void onDestroy() {
```

Comment:
The chromium codes remove observers in `onDestroy` too. I read the documentations and I think the reasoning is like this: the stackoverflow question you linked is about saving data in `onDestroy`, which will cause incomplete data if it is killed, but removing object from list shall be fast enough so that practically it will always be executed.
The reason it cannot be in `onStop` is because `onStop` is called even the activity is still in background, thus if the observer is removed there it will be useless, so `onDestroy` is the only place.

Reply:
yeah as @wchen342 mentioned I see examples in chromium where they remove observers in `onDestroy`. And `onDestroy` mostly not called when the whole process is killed because of low resources and etc and we are good there as we don't save anything.